File: loans.R
data <- read.csv("data/loans.csv")#loading data

#importing libraries
library("tidyr")
library("devtools")

#combining columns 6-25
data <- gather(data, "code", "check", 6:25)

#making separate columns for months and stutus
data <- separate(data, code, c("months", "status"))

#removing irrelevant rows
data <- data[data$check=="X", ]

#removing X in front of months
data$months<-gsub("X","",as.character(data$months))

#dropping the check column
data <- data[ -c(8) ]

#changing the data type of date
data$date <- as.Date(data$date) 


  
data$loan_status <- ifelse((data$status == 'A') | (data$status == 'B'),"current", "expired") #making new columns based on status

data$loan_default <- ifelse((data$status == 'B') | (data$status == 'D'),TRUE , FALSE)

data <- subset(data, select = -c(status)) #removing status as its not needed


#converting the modified data to a csv file
write.csv(data, "loans_r.csv", row.names=FALSE)

File: loans.py
import pandas as pd
df = pd.read_csv('data/loans.csv') #loading data in the variable df
df = df.melt(id_vars=["id", "account_id", "date", "amount", "payments"]) #using melt to combine the columns except id, account_id,date,amount,payments
df = df[df['value'] == 'X'] #dropping extra rows
df['months'], df['status'] = df['variable'].str.split('_', 1).str #creating new columns by splitting 'variable'
df = df.drop(['variable', 'value'], axis=1) #dropping variable and value


def status(row):
    if row['status'] == 'A' or row['status'] == 'B':
        val = 'expired'
    else:
        val = 'current'
    return val

def default(row):
    if row['status'] == 'B' or row['status'] == 'D':
        val = True
    elif row['status'] == 'A' or row['status'] == 'C':
        val = False

    return val

df['loan_status'] = df.apply(status, axis=1)

df['loan_default'] = df.apply(default, axis=1)

df= df.drop(['status'], axis=1)



df.to_csv('loans_py.csv', index=False, encoding='utf-8') #convert to csv